daemon/algod/api/client/restClient.go

```diff
@@ -229,7 +230,7 @@ func (client RestClient) submitForm(
 	}
 
 	if decodeJSON {
-		dec := json.NewDecoder(resp.Body)
+		dec := protocol.NewJSONDecoder(resp.Body)
```

Comment:
Changed `RestClient` to use the `protocol` package's JSON decoding instead of the default Go `json` package's.

The default `json` package does not support our codec tags enough to successfully decode state deltas (which are rather complex). Besides, we encode all REST responses with `protocol`, so it only makes sense to be consistent here.